### PR TITLE
Alert old emr

### PR DESCRIPTION
**Pull request by mshitrit**
/wip

### CODE REVIEW

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -881,3 +885,29 @@ func getIntOrPercentValue(intOrStr *intstr.IntOrString) (int, bool, error) {
 	}
 	return 0, false, fmt.Errorf("invalid type: neither int nor percentage")
 }
+
+func (r *ReconcileMachineHealthCheck) isAlertOldEmr(emr *unstructured.Unstructured) bool {
```
[the remark this answers is not in the document]
Comment (mshitrit) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller.go:889]: IIRC the "is" prefix is quite common for bool type methods/variables.
Not to say it's wrong but I haven't encountered the same usage for "should/was".
There is a very good chance that I'm completely wrong here... and I'm completely out of sync with current conventions :smile: 
Can you point me to anything to support this usage, or do you think that in this case using "should" is just common sense ?

[the remark this answers is not in the document]
Comment (n1r1) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller.go:889]: I wasn't relying on any formal source, just my personal view.
for example, `isAlertedSent` doesn't sound right to me, but `wasAlertSent` sounds better. 

Apple's coding guidelines allow `should` etc.:
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html#//apple_ref/doc/uid/20001282-BCIGIJJF

But since this is just a convention I'm sure there will be other sources saying this is discouraged :)

the bottom line - do whatever you think fit, as long as the repo maintainers allow it.

**Comment by n1r1** [comment]
@mshitrit what will trigger the reconcile function after 2 hours from EMR creation?
don't we need to ask for a requeue?

**Comment by mshitrit** [comment]
> @mshitrit what will trigger the reconcile function after 2 hours from EMR creation?
> don't we need to ask for a requeue?

That's a great question. 
As far as I remember in my tests I did see it was requeued - I'm not sure how though.
I'll look into it and update.

**Comment by mshitrit** [comment]
> @mshitrit what will trigger the reconcile function after 2 hours from EMR creation?
> don't we need to ask for a requeue?

Fixed as we've discussed - setting the default time for next reconcile in case external remediation

**Comment by mshitrit** [comment]
@JoelSpeed , @elmiko 
Can this PR be closed or is it still relevant ?

/cc @slintes

**Comment by JoelSpeed** [comment]
Do we have a Jira card that describes this work? The PR itself looks interesting but I'd be keen to see the motivation behind it. Looks like this could be a useful metric to expose and alert on

**Comment by mshitrit** [comment]
> Do we have a Jira card that describes this work? The PR itself looks interesting but I'd be keen to see the motivation behind it. Looks like this could be a useful metric to expose and alert on

Awhile ago I've tried to push this PR [upstream](https://github.com/openshift/machine-api-operator/pull/852) , ultimately it was rejected.
However IIRC @elmiko created a ticket during this process - I think it might be relevant.

**Comment by JoelSpeed** [comment]
> Awhile ago I've tried to push this PR upstream , ultimately it was rejected.

The link you've included points to this PR, did you mean to point to a cluster api PR?

**Comment by mshitrit** [comment]
> > Awhile ago I've tried to push this PR upstream , ultimately it was rejected.
> 
> The link you've included points to this PR, did you mean to point to a cluster api PR?

Indeed.
Fixed :-)

**Comment by JoelSpeed** [comment]
Left a comment on the issue upstream, let's see if we can have some conversation about the topic there first and come to conclusion on how to proceed based on this
